src/uct/ib/rc/accel/rc_mlx5_ep.c

```diff
@@ -551,6 +553,68 @@ uct_rc_mlx5_ep_check(uct_ep_h tl_ep, unsigned flags, uct_completion_t *comp)
     return UCS_OK;
 }
 
+static ucs_status_t uct_rc_mlx5_ep_check_cb(uct_pending_req_t *self)
+{
+    uct_rc_pending_req_t *req = ucs_derived_of(self, uct_rc_pending_req_t);
+    uct_rc_mlx5_ep_t *ep      = ucs_derived_of(req->ep, uct_rc_mlx5_ep_t);
+    ucs_status_t status;
+
+    ucs_assert(ep->super.flags & UCT_RC_EP_FLAG_KEEPALIVE_POSTED);
+
+    status = uct_rc_mlx5_ep_check_internal(req->ep);
+    if (status == UCS_OK) {
+        ep->super.flags &= ~UCT_RC_EP_FLAG_KEEPALIVE_POSTED;
+        ucs_mpool_put(req);
+    }
+
+    return status;
+}
+
+ucs_status_t
+uct_rc_mlx5_ep_check(uct_ep_h tl_ep, unsigned flags, uct_completion_t *comp)
+{
+    UCT_RC_MLX5_EP_DECL(tl_ep, iface, ep);
+    uct_rc_pending_req_t *req;
+    ucs_status_t status;
+
+    UCT_CHECK_PARAM(comp == NULL, "Unsupported completion on ep_check");
+    UCT_CHECK_PARAM(flags == 0, "Unsupported flags: %x", flags);
+
+    if (ep->super.flags & UCT_RC_EP_FLAG_KEEPALIVE_POSTED) {
```

Comment:
IMO rename to UCT_RC_EP_FLAG_KEEPALIVE_PENDING or UCT_RC_EP_FLAG_KEEPALIVE_QUEUED
to not confuse with post_send